test/features/discord.feature

```diff
@@ -1,101 +1,112 @@
+            | socials  | {"discord":"khattakdev","github":"khattakdev","linkedin":"khattakdev","twitter":"khattakdev"} |
+        Then the response should contain:
+            | documentId | "TYPE:ID" |
+        When make a GET request to "/discord"
+        Then the response status code should be 200
+        And the response in item where field "username" is equal to "khattakdev" should contain:
+            | bio       | "This is a GitHub Campus Expert"                                                              |
+            | username  | "khattakdev"                                                                                  |
+            | socials   | {"discord":"khattakdev","github":"khattakdev","linkedin":"khattakdev","twitter":"khattakdev"} |
+            | updatedOn | "TYPE:DATE"                                                                                   |
+            | createdOn | "TYPE:DATE"                                                                                   |
 
+    Scenario: add an empty user
+        Given authorisation
+        And make a POST request to "/discord" with:
+            | test | "test" |
+        Then the response status code should be 400
+        And the response should contain:
+            | statusCode | 400           |
+            | error      | "Bad Request" |
+        And the response property "message" has items:
+            | username must be a string    |
+            | username should not be empty |
 
-#   Scenario: update a user
-#     Given authorisation
-#     And make a POST request to "/discord" with:
-#       | bio      | "Update user"             |
-#       | username | "update-user"             |
-#       | socials  | {"discord":"update-user"} |
-#     When make a PUT request to "/discord/123" with:
-#       | bio      | "Updated user bio"        |
-#       | username | "updated-user"            |
-#       | socials  | {"discord":"update-user"} |
-#     Then the response status code should be 200
-#     And the response should contain:
-#       | id        | 123                        |
-#       | bio       | "Updated user bio"         |
-#       | username  | "updated-user"             |
-#       | socials   | {"discord":"update-user"}  |
-#       | updatedOn | "2021-01-01T00:00:00.000Z" |
-#       | createdOn | "2021-01-01T00:00:00.000Z" |
+    Scenario: update a user
+        Given authorisation
+        And make a POST request to "/discord" with:
+            | bio      | "This is a GitHub Campus Expert"                                                              |
+            | username | "khattakdev"                                                                                  |
+            | socials  | {"discord":"khattakdev","github":"khattakdev","linkedin":"khattakdev","twitter":"khattakdev"} |
+        Then the response should contain:
+            | documentId | "TYPE:ID" |
+        When make a PUT request to "/discord/{id}" with:
```

Comment:
what is the value of `{id}`?

Reply:
`id` is the Id of the document that got created in the step before. This needs to be dynamically

Reply:
Ah yes, the step def checks for this 👍